pandas/tests/frame/test_constructors.py

```diff
@@ -1933,9 +1933,18 @@ def test_from_records_empty_with_nonempty_fields_gh3682(self):
 
         b = np.array([], dtype=[('id', np.int64), ('value', np.int64)])
         df = DataFrame.from_records(b, index='id')
-        tm.assert_index_equal(df.index, Index([], name='id'))
+        tm.assert_index_equal(df.index, Index([], name='id', dtype='int'))
         assert df.index.name == 'id'
 
+    def test_from_records_empty_dtypes(self):
```

Comment:
Think we should cover more dtypes? Perhaps a good use case for a shared fixture?

Reply:
can you add some more here?